cmd/gardenlet/app/app.go

```diff
@@ -438,6 +457,139 @@ func (g *garden) Start(ctx context.Context) error {
 	return nil
 }
 
+// TODO(aaronfern): Remove this code after v1.91 has been released.
+func (g *garden) createNewDWDResources(ctx context.Context, seedClient client.Client) error {
+	namespaceList := &corev1.NamespaceList{}
+	if err := seedClient.List(ctx, namespaceList, client.MatchingLabels(map[string]string{v1beta1constants.GardenRole: v1beta1constants.GardenRoleShoot})); err != nil {
```

Comment:
For workerless clusters DWD does not start any probe. Can we therefore skip workerless shoot namespaces?

Reply:
For such clusters, [this check](https://github.com/gardener/gardener/pull/9072/files/b56d7e427928fb6b805045baaa4c753da26f7a28#diff-eb33ba906d976101dfe4f69a634d451fd48618d3bfdf57fe7401aca9e03b9bd8R474-R479) should result in an early exit, so we are good, aren't we? Other than this, I'm not aware to identify a workerless shoot just via the namespace. What do you have in mind for the filtering?

Reply:
AFAIK the DWD secrets are still created for workerless clusters. This was a gap identified and acknowledged by @shafeeqes

Reply:
Interesting, correct, I see it in the flow. But then it shouldn't matter whether this migration code is adapted or not - the next shoot reconciliation will create/update the DWD secrets anyways. Isn't this a separate concern to look at (beyond the scope of this PR)?

Reply:
Yes this is a concern that should be addressed in a separate PR (in another communication on slack with blr core colleagues) we came the same conclusion.
I was just trying to optimize, but it seems that there is no way currently to differentiate a workerless shoot just by looking at the namespace labels.